### ISSUE TITLE
A model marked with @versioned cannot be deleted.

### ISSUE THREAD

**Issue by lesmuc** [aws-amplify/amplify-ios#293]
**Describe the bug**
A model marked with @versioned cannot be deleted.

**Version**
0.10.0

**To Reproduce**
```
type Todo @model @versioned {
	id: ID!
	name: String!
	description: String
}
```

Amplify.API.mutate(of: todo, type: .delete) {}

**Expected behavior**
The element should be deleted.

**Result / Error**
Got failed result with GraphQL service returned a successful response containing errors: [Amplify.GraphQLError(message: "Variable \'input\' has coerced Null value for NonNull type \'Int!\'", locations: Optional([Amplify.GraphQLError.Location(line: 1, column: 27)]), path: nil, extensions: nil)]

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version. v12.14.1

Without `@versioned` it worked

**Comment by drochetti** [comment]
Hey @lesmuc we just released a stable [v1.0.0](https://github.com/aws-amplify/amplify-ios/releases/tag/v1.0.0). Could you try?

Two notes about your issue:

**1. Incompatibility between `@versioned` and codegen:**

Can you try adding an explicit `version: Int!` to your GraphQL schema? I suspect this is an issue with the model generator not working properly with `@versioned`.

**2. API breaking change in 1.0.0:**

In `1.0.0` the Amplify.API changed slightly when using models:
```swift
// before
Amplify.API.mutate(of: todo, type: .delete) {}

// after
Amplify.API.mutate(request: .delete(todo)) {}
```

I appreciate your patience and the fact you tried our unstable preview and provided valuable feedback and insights with your issue report.

We also launched a Discord community, feel free to join us there: https://discord.gg/Px5YVfQ

**Comment by drochetti** [comment]
This issue is stale because it had no activity in 30 days. Please re-open if you're still experiencing the same issue.
